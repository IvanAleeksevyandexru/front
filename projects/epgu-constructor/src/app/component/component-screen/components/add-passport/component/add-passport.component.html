<epgu-constructor-screen-pad class="screen-pad-body">
  <section class="add-passport-screen">
    <div class="add-passport-screen__body">
      <epgu-constructor-passport
        [attrs]="$any(data)"
      >
      </epgu-constructor-passport>
    </div>
  </section>
</epgu-constructor-screen-pad>
