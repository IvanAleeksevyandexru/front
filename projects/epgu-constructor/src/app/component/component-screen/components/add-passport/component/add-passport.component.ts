import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { ComponentBase } from '../../../../../screen/screen.types';

@Component({
  selector: 'epgu-constructor-add-passport',
  templateUrl: './add-passport.component.html',
  styleUrls: ['./add-passport.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AddPassportComponent {
  @Input() data: ComponentBase;
}
