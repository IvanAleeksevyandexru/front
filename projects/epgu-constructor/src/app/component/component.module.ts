import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { SharedModule } from '../shared/shared.module';


/**
 * Домен компанент. Сдесь храниться всё что связанно со компанентами и их обязками.
 *
 * TODO: Сюда будем переносить компаненты из домена скринов
 */
@NgModule({
  imports: [
    CommonModule,
    SharedModule
  ],
})
export class ComponentModule {}
