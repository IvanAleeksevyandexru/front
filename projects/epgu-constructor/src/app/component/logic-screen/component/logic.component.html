<div class="wrapper" *ngIf="isLoading$ | async">
  <lib-throbber-hexagon size="big"></lib-throbber-hexagon>
</div>
