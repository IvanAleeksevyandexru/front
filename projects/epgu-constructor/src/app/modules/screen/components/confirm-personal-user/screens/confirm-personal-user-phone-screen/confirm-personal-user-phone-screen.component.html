<epgu-constructor-confirm-personal-user-screen-layout>
  <div class="body">
    <epgu-constructor-confirm-personal-user-phone
      [label]="data?.label"
      [data]="data?.value"
      [error]="errors && errors[data?.id]"
      [isEditButtonShown]="data?.attrs?.actions?.length === 1"
    ></epgu-constructor-confirm-personal-user-phone>
  </div>

  <div class="footer">
    <ng-container #actions *ngIf="data?.attrs?.actions?.length > 1">

    <epgu-constructor-confirm-personal-user-button
      class="confirm-personal-user-button"
      *ngFor="let action of data?.attrs?.actions"
      [action]="action.method"
      (tap)="clickToAction(action)">
      {{ action.label }}
    </epgu-constructor-confirm-personal-user-button>
    </ng-container>
  </div>
</epgu-constructor-confirm-personal-user-screen-layout>
