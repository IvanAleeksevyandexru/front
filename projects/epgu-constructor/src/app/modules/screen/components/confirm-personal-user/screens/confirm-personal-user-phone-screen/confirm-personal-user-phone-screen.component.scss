@import "src/styles/core";

.footer {
  @include tablets-up {
    display: none;
  }
}
