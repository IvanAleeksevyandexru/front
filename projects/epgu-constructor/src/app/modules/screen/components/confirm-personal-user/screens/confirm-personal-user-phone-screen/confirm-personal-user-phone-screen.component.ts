import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
import { EgpuResponseComponentInterface } from '../../../../../../../interfaces/epgu.service.interface';

@Component({
  selector: 'app-confirm-personal-user-phone-screen',
  templateUrl: './confirm-personal-user-phone-screen.component.html',
  styleUrls: ['./confirm-personal-user-phone-screen.component.scss'],
})
export class ConfirmPersonalUserPhoneScreenComponent implements OnInit {
  @Input() data: EgpuResponseComponentInterface;
  @Output() actionSelect = new EventEmitter();

  ngOnInit(): void {}

  clickToAction(action): void {
    this.actionSelect.emit(action);
  }
}
