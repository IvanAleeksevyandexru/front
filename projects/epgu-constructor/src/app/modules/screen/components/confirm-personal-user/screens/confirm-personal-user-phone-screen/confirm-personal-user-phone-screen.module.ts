import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { EpguLibModule } from 'epgu-lib';
import { SharedModule } from '../../../../../../shared-module/shared-components.module';
import { SubComponentsModule } from '../../sub-components/sub-components.module';
import { ConfirmPersonalUserPhoneComponent } from './components/confirm-personal-user-phone/confirm-personal-user-phone.component';
import { ConfirmPersonalUserPhoneScreenComponent } from './confirm-personal-user-phone-screen.component';

const COMPONENTS = [
  ConfirmPersonalUserPhoneScreenComponent,
  ConfirmPersonalUserPhoneComponent,
];

@NgModule({
  declarations: [...COMPONENTS],
  exports: [...COMPONENTS],
  imports: [
    CommonModule,
    SharedModule,
    SubComponentsModule,
    EpguLibModule.forChild(),
  ]
})
export class ConfirmPersonalUserPhoneScreenModule { }
