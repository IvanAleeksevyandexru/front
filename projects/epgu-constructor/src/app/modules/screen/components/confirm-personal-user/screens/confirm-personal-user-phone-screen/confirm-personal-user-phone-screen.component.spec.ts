import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ConfirmPersonalUserPhoneScreenComponent } from './confirm-personal-user-phone-screen.component';



describe('ConfirmPersonalUserPhoneScreenComponent', () => {
  let component: ConfirmPersonalUserPhoneScreenComponent;
  let fixture: ComponentFixture<ConfirmPersonalUserPhoneScreenComponent>;
  const mockData: ComponentInterface = {
    attrs: {},
    id: '',
    label: '',
    type: '',
    value: ''
  };

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      schemas: [ CUSTOM_ELEMENTS_SCHEMA ], // TODO: remove this line when resolve issue with @ifc/plugin and @ifc/common dependencies
      declarations: [ ConfirmPersonalUserPhoneScreenComponent ],
      providers: [ ScreenComponentService ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ConfirmPersonalUserPhoneScreenComponent);
    component = fixture.componentInstance;
    component.data = mockData;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
