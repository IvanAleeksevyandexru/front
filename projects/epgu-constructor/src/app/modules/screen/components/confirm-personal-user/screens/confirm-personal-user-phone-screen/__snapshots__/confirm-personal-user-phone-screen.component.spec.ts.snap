// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmPersonalUserPhoneScreenComponent check snapshot 1`] = `
<epgu-constructor-confirm-personal-user-phone-screen
  __ngContext__={[Function LRootView]}
  data={[Function Object]}
  nextStepEvent={[Function EventEmitter_]}
  screenComponentService={[Function ScreenComponentService]}
>
  <epgu-constructor-confirm-personal-user-screen-layout>
    <div
      class="body"
    >
      <epgu-constructor-confirm-personal-user-phone />
    </div>
    <div
      class="footer"
    >
      
    </div>
  </epgu-constructor-confirm-personal-user-screen-layout>
</epgu-constructor-confirm-personal-user-phone-screen>
`;
